packages/react/src/components/Authenticator/FederatedSignIn/FederatedSignInButtons/FederatedSignInButton.tsx

```diff
@@ -1,15 +1,18 @@
 import { useAmplify, useAuth } from '../../../../hooks';
 import { FederatedIdentityProviders } from '@aws-amplify/ui';
 
+import { IconFacebook } from '../../../../primitives';
+
 export interface FederatedSignInButtonProps {
+  icon?: 'facebook' | 'google' | 'amazon';
```

Comment:
I think we'll be revisiting this implementation and how it's exposed for customers to customize via slots (Vue/Angular) or provider (React/RN), so start thinking about what abstraction makes the most sense for customers to replace/inject OAuth/SAML providers (e.g. Auth0, Discord, GitHub, Twitter) in the UI.

Reply:
Sure, this is just to get us started for now.